lib/CL/clReleaseKernel.c

```diff
@@ -37,29 +37,32 @@ POname(clReleaseKernel)(cl_kernel kernel) CL_API_SUFFIX__VERSION_1_0
   if (new_refcount == 0)
     {
       POCL_MSG_PRINT_INFO ("Freeing kernel %p\n", kernel);
-      if (kernel->program != NULL)
+      cl_program program = kernel->program;
+      if ((program != NULL)
+          && (!program->operating_on_default_kernels))
```

Comment:
Well, this is an improvement, but not a huge one, to the previous magic pointer values :) This branch at least needs comments.